File: client/src/boundary/model/BoundaryShip.ts
export default class BoundaryShip {
  readonly length: number;
  readonly coordinates: { x: number; y: number }[];
  readonly hits: number;
  readonly destroyed: boolean;

  constructor(
    length: number,
    coordinates: { x: number; y: number }[],
    hits: number,
    destroyed: boolean
  ) {
    this.length = length;
    this.coordinates = coordinates;
    this.hits = hits;
    this.destroyed = destroyed;
  }
}

File: server/battleships/use_cases/api/src/main/java/eu/optas/use_cases/api/BoundaryShip.java
package eu.optas.use_cases.api;

import eu.optas.utils.Coordinates;

import java.util.List;

public class BoundaryShip {
    private final int length;
    private final List<Coordinates> coordinates;
    private final int hits;
    private final boolean destroyed;

    public BoundaryShip(int length, List<Coordinates> coordinates, int hits, boolean destroyed) {
        this.length = length;
        this.coordinates = coordinates;
        this.hits = hits;
        this.destroyed = destroyed;
    }

    public int getLength() {
        return length;
    }

    public List<Coordinates> getCoordinates() {
        return coordinates;
    }

    public int getHits() {
        return hits;
    }

    public boolean isDestroyed() {
        return destroyed;
    }
}
